pipeline.wdl

```diff
@@ -0,0 +1,246 @@
+            outputPrefix = outputDirectory + "/" + pipelineRunName,
+            genomeBuild = genomeBuild,
+            annotationVersion = annotationVersion,
+            filterTranscripts = filterTranscriptsAbundance,
+            filterPairingsFile = filterPairingsFileAbundance,
+            dockerImage = dockerImages["talon"]
+    }
+
+    call talon.SummarizeDatasets as createSummaryFile {
+        input:
+            databaseFile = runTalon.outputUpdatedDatabase,
+            outputPrefix = outputDirectory + "/" + pipelineRunName,
+            datasetGroupsCSV = summaryDatasetGroupsCSV,
+            dockerImage = dockerImages["talon"]
+    }
+
+    output {
+        File outputSpliceJunctionsFile = select_first([spliceJunctionsFile, createSJsfile.outputSJsFile])
+        Array[File] outputMinimap2 = flatten(sampleWorkflow.outputMinimap2)
+        Array[File] outputTranscriptCleanFasta = flatten(sampleWorkflow.outputTranscriptCleanFasta)
+        Array[File] outputTranscriptCleanLog = flatten(sampleWorkflow.outputTranscriptCleanLog)
+        Array[File] outputTranscriptCleanSAM = flatten(sampleWorkflow.outputTranscriptCleanSAM)
+        Array[File] outputTranscriptCleanTElog = flatten(sampleWorkflow.outputTranscriptCleanTElog)
+        File outputTalonDatabase = runTalon.outputUpdatedDatabase
+        File outputTalonLog = runTalon.outputLog
+        File outputAbundance = createAbundanceFile.outputAbundanceFile
+        File outputSummary = createSummaryFile.outputSummaryFile
+    }
+}
+
+task RunTalonOnLoop {
+    input {
+        Array[File] SAMfiles
+        File configFile
+        File databaseFile
+        String outputPrefix
+        String genomeBuild
+        String sequencingPlatform = "PacBio-Sequal"
+
+        Float? minimumCoverage = 0.9
+        Int? minimumIdentity = 0
+
+        Int cores = 1
+        Int memory = 20
+        String dockerImage = "biocontainers/talon:v4.2_cv2"
+    }
+
+    command <<<
+        #$ -l h_vmem=~{memory}G
+        #$ -pe BWA ~{cores}
+        set -e
+        mkdir -p $(dirname ~{outputPrefix})
+        counter=1
+        for file in ~{sep=" " SAMfiles}
+        do
+            userInput="$(sed -n ${counter}p ~{configFile})"
+            configFileLine="${userInput},~{sequencingPlatform},${file}"
+            echo ${configFileLine} > configFile.csv
+            talon \
+                --f configFile.csv \
+                ~{"--db " + databaseFile} \
+                ~{"--o " + outputPrefix} \
+                ~{"--build " + genomeBuild} \
+                ~{"--cov " + minimumCoverage} \
+                ~{"--identity " + minimumIdentity}
+            counter=$((counter+1))
+        done
+    >>>
+
+    output {
+        File outputUpdatedDatabase = databaseFile
+        File outputLog = outputPrefix + "_talon_QC.log"
+    }
+
+    runtime {
+        cpu: cores
+        memory: memory
+        docker: dockerImage
+    }
+}
+
+task SampleConfigToSampleReadgroupLists {
```

Comment:
As a general note, it might be better to move this to tasks/common.wdl

Reply:
I copied this from the small-rna pipeline, but we could indeed move it to tasks/common.wdl.